File: StripAccents.java
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.text.Normalizer;

public class StripAccents {
    public static void main(String[] args) {
        String filePath = "JogoAudrey.java";
        Path path = Paths.get(filePath);

        try {
            String content = new String(Files.readAllBytes(path), StandardCharsets.UTF_8);

            content = stripAccents(content);

            Files.write(path, content.getBytes(StandardCharsets.UTF_8));
            System.out.println("Accents stripped.");

        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static String stripAccents(String text) {
        String normalized = Normalizer.normalize(text, Normalizer.Form.NFD);
        return normalized.replaceAll("\\p{M}", "");
    }
}

File: strip_accents.py
import sys
import re
import unicodedata

file_path = "JogoAudrey.java"

with open(file_path, "r", encoding="utf-8") as f:
    content = f.read()

# Substituir manualmente os caracteres acentuados utf-8 para seus equivalentes ascii
def strip_accents(text):
    return ''.join(c for c in unicodedata.normalize('NFD', text)
                  if unicodedata.category(c) != 'Mn')

# Apenas substituir na parte de texto (ou em tudo, já que é pt-br)
content = strip_accents(content)

with open(file_path, "w", encoding="utf-8") as f:
    f.write(content)

print("Accents stripped.")
